tests/e2e/framework/testInfo.go

```diff
@@ -184,7 +184,14 @@ func (t testInfo) FetchAndSaveClusterLogs(namespace string) error {
 			}
 			// append logs to file
 			for _, logEntry := range entries {
-				_, err = f.WriteString(fmt.Sprintf("%v\n", logEntry.Payload))
+				timestamp := logEntry.GetTimestamp()
+				fmtTime := time.Unix(timestamp.Seconds, 0)
+				log := fmt.Sprintf("[%02d:%02d:%02d] %s",
+					fmtTime.Hour(), fmtTime.Minute(), fmtTime.Second(), logEntry.GetTextPayload())
+				if log[len(log)-1:] != "\n" {
+					log += "\n"
```

Comment:
Yeah I was amazed by the heterogeneity in logs. The payload is formatted different and some has newline and the end while some don't. I do see the point of taking advantage of the file write buffer since concatenation on string essentially claims additional memory and copy everything over. I did take a look on the WriteString() source code and see no locks. I assume copy-on-write until we close the file to commit our writes.